src/crimson/os/seastore/seastore_perf_counters.h

```diff
@@ -0,0 +1,75 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+#pragma once
+
+#include "crimson/common/perf_counters_collection.h"
+
+namespace crimson::os::seastore {
+
+using PerfCountersRef = std::unique_ptr<PerfCounters>;
+enum {
+  seastore_perfcounters_first = 30000,
+  ss_tm_extents_mutated_total,
+  ss_tm_extents_mutated_bytes,
+  ss_tm_extents_retired_total,
+  ss_tm_extents_retired_bytes,
+  ss_tm_extents_allocated_total,
+  ss_tm_extents_allocated_bytes,
+  ss_sm_bytes_written,
+  ss_sm_segments_released,
+  ss_sm_bytes_released,
+  seastore_perfcounters_last,
+};
+
+class PerfService {
+  void build_seastore_perf() {
```

Comment:
This should return a unique_ptr<PerfCounters> and be used to directly initialize ss_perf in the constructor colon initialization list.

Reply:
I think all parts in seastore use one perfcounter,  so it build once and save to PerfService's ss_perf and each part use reference to this perfcounter.  do you mean each part such as transaction_manager and segment_manager build their own perfcounter and add them to perfcounters_collection?  ( each part does same add_to_collection and remove_to_collection?)

Reply:
No, I mean the ss_perf member should be a unique_ptr as I mentioned above and you should invoke this function in PerfService() to initialize it.